pages/software/python.md

```diff
@@ -0,0 +1,457 @@
+!!! Note  
+    Not all users have access to a shared project directory.  Please verify your access via [ColdFront](../portals/coldfront.md).  
+
+## Using CCR's Python Installation  
+
+CCR provides as part of the currently supported toolchains.  We do not recommend using a version included within the operating system nor the one provided in CCR's compatibility layer.  These can differ across login and compute nodes and can be replaced with newer versions, potentially affecting your project.  These are also compiled to make the best use out of the architecture you're running on rather than offering one solution for the lowest common denominator.  Rather, using one contained in CCR's software repository toolchain provides you the most flexibility and usability across all CCR systems.  We highly recommend using python virtual environments for your projects.  A python virtual environment can be created using a specific version of python and can be used in isolation of other environments.  This is convenient if your projects require different versions of Python packages or have differing dependencies.  Within the virtual environment you can install additional Python packages  These virtual environments can be stored in your home directory or within your group's project directory making them available to other members of your group.  
+
+
+### Python software modules
+
+Find and display the available Python modules using the `module spider` command, then load the version of Python you'd like to use.  If you have no requirements, we recommend loading the latest.  The spider output will tell you if there are other modules you must load prior to loading your selected python. For example:  
+```
+ccruser@vortex ~ $ module spider python  
+------------------------------------------------------------------------------------------------------------
+  python:
+------------------------------------------------------------------------------------------------------------
+    Description:
+      Python is a programming language that lets you work more quickly and integrate your systems more
+      effectively.
+
+     Versions:
+        python/3.9.5-bare
+        python/3.9.5
+        python/3.9.6-bare
+        python/3.9.6
+     Other possible modules matches:
+        ipython
+
+ccruser@vortex ~ $ module spider python/3.9.6    
+------------------------------------------------------------------------------------------------------------
+  python: python/3.9.6
+------------------------------------------------------------------------------------------------------------
+    Description:
+      Python is a programming language that lets you work more quickly and integrate your systems more
+      effectively.
+
+    Properties:
+      Tools for development
+
+    You will need to load all module(s) on any one of the lines below before the "python/3.9.6" module is available to load.
+
+      gcccore/11.2.0
+
+    This module provides the following extensions:
+
```

Comment:
would suggest cutting this down a bit.. no need to show the extensions here in the example module output

Reply:
removed